ceph/rados/pool_workflows.py

```diff
@@ -70,3 +70,53 @@ def get_pg_autoscaler_value(self, pool_name, item):
         except Exception:
             log.error("Pool not found")
         return pool_details[item]
+
+    def fill_omap_entries(self, pool_name, **kwargs):
+        """
+        creates key-value entries for objects on ceph pools and increase the omap entries on the pool
+        eg : if obj_start, obj_end: 0, 3000 objects, with num_keys 1000,  the method would create 3000 objects with 1k
+        KW pairs each. so total 3000*1000 KW entries
+        Args:
+            pool_name: name of the pool where the KW pairs needed to be added to objects
+            **kwargs: other args that can be passed
+                Valid args:
+                1. obj_start: start count for object creation
+                2. obj_end : end count for object creation
+                3. num_keys_obj: Number of KW paris to be added to each object
+
+        Returns: True -> pass, False -> fail
+        """
+        # Getting the client node to perform the operations
+        client_node = self.rados_obj.ceph_cluster.get_nodes(role="client")[0]
+        obj_start = kwargs.get("obj_start", 0)
+        obj_end = kwargs.get("obj_end", 2000)
+        num_keys_obj = kwargs.get("num_keys_obj", 20000)
+        log.debug(
+            f"Writing {(obj_end - obj_start) * num_keys_obj} Key paris"
+            f" to increase the omap entries on pool {pool_name}"
+        )
+        script_loc = "https://raw.githubusercontent.com/red-hat-storage/cephci/master/utility/generate_omap_entries.py"
+        client_node.exec_command(
+            sudo=True,
+            cmd=f"curl -k {script_loc} -O",
+        )
+        cmd_options = f"--pool {pool_name} --start {obj_start} --end {obj_end} --key-count {num_keys_obj}"
+        cmd = f"python3 generate_omap_entries.py {cmd_options}"
+        client_node.exec_command(sudo=True, cmd=cmd, long_running=True)
+
+        log.debug("Checking the amount of omap entries created on the pool")
+        pool_stats = self.rados_obj.run_ceph_command(cmd="ceph df detail")["pools"]
+        for detail in pool_stats:
+            if detail["name"] == pool_name:
+                pool_1_stats = detail["stats"]
+                total_omap_data = pool_1_stats["omap_bytes_used"]
+                omap_data = pool_1_stats["stored_omap"]
+                break
+        if omap_data < 0:
+            log.error("No omap entries written into pool")
+            return False
+        log.info(
+            f"Wrote {omap_data} bytes of omap data on the pool."
+            f"Total stored omap data on pool : {total_omap_data}"
+        )
+        return True
```

Comment:
Cleaning up the generate_omap_entries.py file in the driver node

Reply:
added step to remove the file created